# Milestone 2

### Name    : Gracia Valerine

### Batch   : HCK 017

### This program is build to predict the price of houses based on the datas that has been gathered.

# Model Inference

### Import Libraries

In [10]:
# Import Library

#Dataframe
import pandas as pd

# Numerical Data
import numpy as np

# Save Model
import pickle
import json

In [11]:
with open('list_num_cols.txt', 'r') as file_1:
  list_num_col = json.load(file_1)

with open("scaler.pkl", "rb") as file_2:
  scaler = pickle.load(file_2)

with open("model.pkl", "rb") as file_3:
  rf = pickle.load(file_3)

In [12]:
#Data Dummy
dataDummy = {
    'bedrooms': [4, 1, 5, 5, 1, 6, 6, 5, 5, 5],
    'bathrooms': [1.2, 2.1, 3.3, 3.7, 3.8, 4.7, 2.1, 3.2, 3.7, 4.3],
    'sqft_living': [3236, 4184, 2463, 4524, 1714, 1638, 1810, 3704, 4743, 3356],
    'sqft_lot': [12961, 7689, 6273, 11680, 5227, 10835, 9115, 3213, 14020, 5611],
    'floors': [2.5, 2.3, 2.1, 1.7, 2.4, 1.7, 2.1, 2.1, 1.0, 1.1],
    'waterfront': [1, 0, 0, 1, 0, 0, 1, 0, 0, 0],
    'view': [2, 0, 1, 4, 4, 1, 1, 2, 0, 1],
    'condition': [3, 4, 3, 4, 1, 4, 3, 4, 3, 1],
    'grade': [11, 12, 7, 6, 6, 2, 3, 8, 1, 12],
    'sqft_above': [1717, 3036, 2049, 769, 3336, 1622, 1172, 915, 1913, 1321],
    'sqft_basement': [1308, 1837, 977, 1811, 1264, 1966, 1089, 1791, 1807, 1544],
    'yr_built': [1950, 1950, 1989, 1938, 1996, 1993, 1972, 1930, 1999, 1990],
    'yr_renovated': [1802, 628, 502, 724, 291, 723, 1623, 486, 1832, 1870],
    'zipcode': [98017, 98520, 98132, 98168, 98257, 98834, 98333, 98619, 98170, 98348],
    'lat': [47.878891, 47.000912, 47.659236, 47.633970, 47.142391, 47.764621, 47.776950, 47.175165, 47.067244, 47.310015],
    'long': [-121.771605, -121.764418, -121.757800, -121.737283, -122.441047, -121.736246, -121.855143, -121.705866, -122.352264, -121.532285],
    'sqft_living15': [4934, 814, 573, 1620, 1201, 800, 3412, 4562, 3754, 4256],
    'sqft_lot15': [11941, 7391, 11911, 2076, 6695, 1151, 4734, 10857, 13988, 1363]
}

#Turn into data frame
dfDataDummy = pd.DataFrame(dataDummy)

#Check data
dfDataDummy.tail()

bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  view  \
5         6        4.7         1638     10835     1.7           0     1   
6         6        2.1         1810      9115     2.1           1     1   
7         5        3.2         3704      3213     2.1           0     2   
8         5        3.7         4743     14020     1.0           0     0   
9         5        4.3         3356      5611     1.1           0     1   

   condition  grade  sqft_above  sqft_basement  yr_built  yr_renovated  \
5          4      2        1622           1966      1993           723   
6          3      3        1172           1089      1972          1623   
7          4      8         915           1791      1930           486   
8          3      1        1913           1807      1999          1832   
9          1     12        1321           1544      1990          1870   

   zipcode        lat        long  sqft_living15  sqft_lot15  
5    98834  47.764621 -121.736246            800        1151  
6    98333  47.776950 -121.855143           3412        4734  
7    98619  47.175165 -121.705866           4562       10857  
8    98170  47.067244 -122.352264           3754       13988  
9    98348  47.310015 -121.532285           4256        1363

These are the data that will be tested to see how well the model performs with different data. 

### Inference Feature Engineering

In [13]:
#Scaling, and concating
data_inference_final = scaler.transform(dfDataDummy)

#Check data
data_inference_final

array([[ 1.00000000e+00, -1.40000000e+00,  1.17345133e+00,
         9.51156584e-01,  1.00000000e+00,  1.00000000e+00,
         2.00000000e+00,  0.00000000e+00,  4.00000000e+00,
         1.53921569e-01,  2.29473684e+00, -5.55555556e-01,
         1.80200000e+03, -5.71428571e-01,  1.47751654e+00,
         2.25317734e+00,  3.52727273e+00,  8.66148835e-01],
       [-2.00000000e+00, -2.00000000e-01,  2.01238938e+00,
         1.30782918e-02,  8.00000000e-01,  0.00000000e+00,
         0.00000000e+00,  1.00000000e+00,  5.00000000e+00,
         1.44705882e+00,  3.22280702e+00, -5.55555556e-01,
         6.28000000e+02,  5.41666667e+00, -2.74810733e+00,
         2.28858128e+00, -1.15454545e+00, -4.59032824e-02],
       [ 2.00000000e+00,  1.40000000e+00,  4.89380531e-01,
        -2.38879004e-01,  6.00000000e-01,  0.00000000e+00,
         1.00000000e+00,  0.00000000e+00,  0.00000000e+00,
         4.79411765e-01,  1.71403509e+00,  3.11111111e-01,
         5.02000000e+02,  7.97619048e-01,  4.20339309e

The data dummy will be scaled for an easier read that will suit the model.

### Inference Prediction

The random forest regressor model will be initialized once more with the best hyperparameter that have been checked by the hyperparameter tuning before. This will hopefully give the best result for the predictions.

In [14]:
#Predict  score
predicted_score = rf.predict(data_inference_final)

#Show result
print(f"Predicted Overall: {predicted_score[0]:.2f}")

Predicted Overall: 366082.12


/opt/anaconda3/lib/python3.11/site-packages/sklearn/base.py:493: UserWarning: X does not have valid feature names, but RandomForestRegressor was fitted with feature names
  warnings.warn(


The data inference that has been predicted the overall resulted as 366082.12. We can see that the result is quite reasonable for the house price. 